File: frontend/src/types/image.ts
/** Mirrors backend/app/schemas/image.py — keep in sync. */

export interface ImageOut {
  id: string;
  filename: string;
  content_type: string;
  size_bytes: number;
  width: number | null;
  height: number | null;
  bounds: [number, number, number, number] | null;
  crs: string | null;
  source: string;
  plant_id: string | null;
  meta: Record<string, unknown>;
  created_at: string;
  url: string | null;
  preview_url: string | null;
}

export interface ImageListOut {
  items: ImageOut[];
  total: number;
  page: number;
  page_size: number;
}

File: backend/app/schemas/image.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class ImageOut(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    filename: str
    content_type: str
    size_bytes: int
    width: int | None
    height: int | None
    bounds: list[float] | None
    crs: str | None
    source: str
    plant_id: uuid.UUID | None
    meta: dict
    created_at: datetime
    # populated by the router, not the ORM
    url: str | None = None
    preview_url: str | None = None


class ImageListOut(BaseModel):
    items: list[ImageOut]
    total: int
    page: int
    page_size: int
